Tpetra:  unused variable 'tfecfFuncName' warning

## Bug Report
@trilinos/tpetra

### Description
Sierra compilation with GCC 7.2 is failing (warnings are treated as errors) because of
```
Tpetra_CrsGraph_def.hpp:2356:16: error: unused variable 'tfecfFuncName' [-Werror=unused-variable]
     const char tfecfFuncName[] = "findLocalIndices: ";
                ^~~~~~~~~~~~~
```

### Steps to Reproduce
1. Checkout Trilinos 7506e6b2909e2dd8841e42471bea2f99346f9a53 into Sierra
1. Build Sierra aria

--- a/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp
+++ b/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp
@@ -2353,8 +2353,8 @@ namespace Tpetra {
                    const Teuchos::ArrayView<const LocalOrdinal>& indices,
                    std::function<void(const size_t, const size_t, const size_t)> fun) const
   {
-    const char tfecfFuncName[] = "findLocalIndices: ";
 #ifdef HAVE_TPETRA_DEBUG
+    const char tfecfFuncName[] = "findLocalIndices: ";
     TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
       (this->getProfileType() != StaticProfile, std::runtime_error,
        "findLocalIndices requires that the graph have StaticProfile.");

